content/docs/projects/csi-driver.md

```diff
@@ -0,0 +1,228 @@
+| `csi.cert-manager.io/certificate-file`  | File name to store the certificate file at.                                                                                | `tls.crt`                            | `bar/foo.crt`                    |
+| `csi.cert-manager.io/ca-file`           | File name to store the ca certificate file at.                                                                             | `ca.crt`                             | `bar/foo.ca`                     |
+| `csi.cert-manager.io/privatekey-file`   | File name to store the key file at.                                                                                        | `tls.key`                            | `bar/foo.key`                    |
+| `csi.cert-manager.io/fs-group`          | Set the FS Group of written files. Should be paired with and match the value of the consuming container `runAsGroup`.      |                                      | `2000`                           |
+| `csi.cert-manager.io/renew-before`      | The time to renew the certificate before expiry. Defaults to a third of the requested duration.                            | `$CERT_DURATION/3`                   | `72h`                            |
+| `csi.cert-manager.io/reuse-private-key` | Re-use the same private when when renewing certificates.                                                                   | `false`                              | `true`                           |
+
+### Variables
+
+The following attributes support variables that are evaluated when a request is
+made for the mounting Pod. These variables are useful for constructing requests
+with SANs that contain values from the mounting Pod.
+
+```
+`csi.cert-manager.io/common-name`
+`csi.cert-manager.io/dns-names`
+`csi.cert-manager.io/uri-sans`
+```
+
+Variables follow the [go `os.Expand`](https://pkg.go.dev/os#Expand) structure,
+which is generally what you would expect on a UNIX shell. The CSI driver has
+access to the following variables:
+
+```
+${POD_NAME}
+${POD_NAMESPACE}
+${POD_UID}
+```
+
+#### Example Usage
+
+```yaml
+volumeAttributes:
+  csi.cert-manager.io/issuer-name: ca-issuer
+  csi.cert-manager.io/dns-names: "${POD_NAME}.${POD_NAMESPACE}.svc.cluster.local"
+  csi.cert-manager.io/uri-sans: "spiffe://cluster.local/ns/${POD_NAMESPACE}/pod/${POD_NAME}/${POD_UID}"
+  csi.cert-manager.io/common-name: "${POD_NAME}.${POD_NAMESPACE}"
+```
+
+## Requesting Certificates using the mounting Pod's ServiceAccount
+
+If the flag `--use-token-request` is enabled on the csi-driver DaemonSet, the
+[CertificateRequest](../concepts/certificaterequest/) resource will be created
+by the mounting Pod's ServiceAccount. This can be pared with
+[approver-policy](./approver-policy/) to enable advanced policy on a per
```

Comment:
this link is broken in the deploy preview

Reply:
This is expected until https://github.com/cert-manager/website/pull/999 merges